code/modules/admin/verbs/playsound.dm

```diff
@@ -52,84 +59,91 @@
+	var/stderr = output[SHELLEO_STDERR]
+	if(!errorlevel)
+		var/list/data
+		try			//NOTE: Try/catch might break admin proccalls on exception catch. It is recommended to admins to not directly call this proc unless they really know what they are doing or are willing to lose their proccall for a round.
+			data = json_decode(stdout)
+		catch(var/exception/e)
+			return YTDL_FAILURE_BAD_JSON
+		if(data["url"] && !findtext(data["url"], GLOB.is_http_protocol))
+			return YTDL_FAILURE_NO_HTTPS
+		return data
+	else
+		if(!dump_error_to_chat)
+			stack_trace("YoutubeDL error: [stderror]")
+		else
+			if(!dump_to && usr)
+				dump_to = usr
+			to_chat(dump_to, "<span class='boldwarning'>Youtube-dl retrieval failed.</span><span class='warning'> Error: [stderror]</span>")
+	return YTDL_FAILURE_UNKNOWN
+
 /client/proc/play_web_sound()
 	set category = "Fun"
 	set name = "Play Internet Sound"
 	if(!check_rights(R_SOUNDS))
 		return
-
-	var/ytdl = CONFIG_GET(string/invoke_youtubedl)
-	if(!ytdl)
+	var/url = input("Enter content URL (supported sites only, leave blank to stop playing)", "Play Internet Sound via youtube-dl") as text|null
+	if(!length(url))//pressed ok with blank
+		log_admin("[key_name(src)] stopped web sound")
+		message_admins("[key_name(src)] stopped web sound")
+		for(var/m in GLOB.player_list)
+			var/mob/M = m
+			var/client/C = M.client
+			if((C.prefs.toggles & SOUND_MIDI) && C.chatOutput && !C.chatOutput.broken && C.chatOutput.loaded)
+				C.chatOutput.stopMusic()
+		SSblackbox.record_feedback("tally", "admin_verb", 1, "Play Internet Sound")
+		return
+	var/list/returned = invoke_youtubedl(url, TRUE, src)
+	if(returned == YTDL_NOT_CONFIGURED)
 		to_chat(src, "<span class='boldwarning'>Youtube-dl was not configured, action unavailable</span>") //Check config.txt for the INVOKE_YOUTUBEDL value
 		return
-
-	var/web_sound_input = input("Enter content URL (supported sites only, leave blank to stop playing)", "Play Internet Sound via youtube-dl") as text|null
-	if(istext(web_sound_input))
-		var/web_sound_url = ""
-		var/stop_web_sounds = FALSE
-		var/pitch
-		if(length(web_sound_input))
-
-			web_sound_input = trim(web_sound_input)
-			if(findtext(web_sound_input, ":") && !findtext(web_sound_input, GLOB.is_http_protocol))
-				to_chat(src, "<span class='boldwarning'>Non-http(s) URIs are not allowed.</span>")
-				to_chat(src, "<span class='warning'>For youtube-dl shortcuts like ytsearch: please use the appropriate full url from the website.</span>")
+	if(returned == YTDL_FAILURE_NO_HTTPS)
+		to_chat(src, "<span class='boldwarning'>Non-http(s) URIs are not allowed.</span>")
+		to_chat(src, "<span class='warning'>For youtube-dl shortcuts like ytsearch: please use the appropriate full url from the website.</span>")
+		return
+	if(returned == YTDL_FAILURE_BAD_JSON)
+		to_chat(src, "<span class='boldwarning'>Youtube-dl JSON parsing FAILED:</span>")
+		to_chat(src, "<span class='warning'>[e]: [stdout]</span>")

```

Comment:
this used to spit out the byond error with parsing
https://travis-ci.org/tgstation/tgstation/jobs/431752482#L621

and stdout of youtube-dl
https://travis-ci.org/tgstation/tgstation/jobs/431752482#L622

could probably be done in line 83